Wire Up Apiary Interaction with Back-end

rajadain opened the pull request:
## Overview

Wires up some of the front-end apiary activities with the back-end.

Connects #340 

### Demo

#### Automatic Logging In

![2018-12-10 17 04 07](https://user-images.githubusercontent.com/1430060/49764697-23c5d480-fc9e-11e8-92b5-5ba148a32850.gif)

#### Saving Apiaries to the Database

![2018-12-10 17 05 01](https://user-images.githubusercontent.com/1430060/49764715-3a6c2b80-fc9e-11e8-8ba2-9f817d14d11f.gif)

#### Starring and Unstarring Apiaries

![2018-12-10 17 05 34](https://user-images.githubusercontent.com/1430060/49764738-46f08400-fc9e-11e8-8e43-5ddf76a4fa91.gif)

#### Deleting Apiaries

![2018-12-10 17 06 03](https://user-images.githubusercontent.com/1430060/49764748-4f48bf00-fc9e-11e8-9cba-ac1157040218.gif)

## Testing Instructions

* Log in. Refresh the screen. Ensure you are logged in.
* Make a few apiaries. Refresh the page. Ensure you see the same apiaries.
* Star some of them. Refresh the page. Ensure they're the same as before.
* Delete some of them. Refresh the page. Ensure they're still deleted.
  * Inspect the database. Ensure the apiaries are still there, just with non-NULL values in `deleted_at`
* Log out by going to [:8000/user/logout](http://localhost:8000/user/logout), and then back to [:8000/?beekeepers](http://localhost:8000/?beekeepers)
* Ensure the making apiaries, starring, unstarring, and deleting work even when you're logged out

rajadain commented:
Updated to target `develop` since #360 was merged.

fungjj92 commented on src/icp/apps/beekeepers/js/src/components/ApiaryCard.jsx:
```diff
@@ -78,9 +83,14 @@ const ApiaryCard = ({ apiary, forageRange }) => {
     );
 };
 
+function mapStateToProps() {
```
We can remove the unused `mapStateToProps` and in the `connect` declaration below validly write:
```
export default connect()(ApiaryCard);
```

rajadain replied:
Ah thank you! I had tried `connect(() => {})(ApiaryCard)` which hadn't worked. This is much neater.

rajadain replied:
Resoled by 90aa3aa5

rajadain commented on src/icp/apps/beekeepers/js/src/App.jsx:
```diff
@@ -8,17 +10,49 @@ import SignUpModal from './components/SignUpModal';
+    componentDidMount() {
+        const { dispatch } = this.props;
+
+        dispatch(attemptLogin());
```
I think the logging in function should be handled at the top-most level. At the time of writing, that was `App`. Since `App` and `Survey` share so much (the `Header` and modal components), perhaps that can be generalized into a common parent, which can also house this logic.

rajadain replied:
I just attended a talk at LibertyJS that strongly recommended using blank templates of content rather than spinners for the load state. I think that's the way we should go. But I also think that's polish that can wait till the end. Let's make a card so we don't forget about it though.

rajadain commented on src/icp/apps/beekeepers/js/src/App.jsx:
```diff
@@ -8,17 +10,49 @@ import SignUpModal from './components/SignUpModal';
+
+App.propTypes = {
+    dispatch: func.isRequired,
+    userId: number.isRequired,
```
Resolved by ca18d2a

rajadain commented on src/icp/apps/beekeepers/js/src/actions.js:
```diff
@@ -129,3 +162,107 @@ export function login(form) {
     };
 }
+
+export function attemptLogin() {
```
Resolved by a0017d20

rajadain commented:
This is ready for another look.

rajadain commented:
I squashed the fixups, and added a commit that updates the `surveyed` state just as `starred` is done. Will merge when green. Thanks for the review and the insights!
